dashboards-observability/public/components/visualizations/charts/bar/bar.tsx

```diff
@@ -68,11 +70,11 @@ export const Bar = ({ visualizations, layout, config }: any) => {
     PLOTLY_COLOR[index % PLOTLY_COLOR.length];
 
   const prepareData = (valueForXSeries) => {
-    return (valueForXSeries.map((dimension: any) => data[dimension.label]))?.reduce(
-      (prev, cur) => {
+    return valueForXSeries
+      .map((dimension: any) => data[dimension.label])
+      ?.reduce((prev, cur) => {
         return prev.map((i, j) => `${i}, ${cur[j]}`);
-      }
-    );
+      });
```

Comment:
As this is going to be a longer array with all the values, can't we combine the map and reduce functions? We can fetch the label inside the reduce function instead of having a separate function to loop over all items just to fetch the labels. 

I understand this may not be your code, can we update this to enhance the code?

Reply:
It has been taken care of in a different branch, hence not modifying the code here. In that branch, we are only using reduce to loop over the data and fetch the label.